test/unit/bitmap_test.c

```diff
@@ -0,0 +1,80 @@
+#include <runtime.h>
+#include <stdlib.h>
+#define EXIT_FAILURE 1
+#define EXIT_SUCCESS 0
+
+#define WHOLE_RANGE irange(0, infinity)
+
+
+/**
+ *  Tests allocation of bitmap using allocate_bitmap 
+ *  function.
+ */
+bitmap test_alloc() {
+    heap h = init_process_runtime();
+    bitmap b = allocate_bitmap(h, h, infinity);
+    if (b == INVALID_ADDRESS) 
+        msg_err("!!! allocation failed for bitmap\n");
+    return b;
```

Comment:
Here you could test that a just-allocated bitmap doesn't have any bit set, e.g. by using the `bitmap_foreach_set` macro

Reply:
@francescolavra I looked at the bitmap_foreach_set macro and its application in poll.c, but I'm still kind of lost on what the second parameter is used for and how I could use it to check each bit. Could you explain its functionality or direct me to somewhere in the codebase that can explain it better?

Reply:
The second parameter of the bitmap_foreach_set macro is just a name or your choice. This macro expands to a `for` loop that has as many iterations as there are bits set in the bitmap, and at each iteration the variable whose name you chose as second parameter will contain the value of a bit set in the bitmap. So you could use this macro to verify that a bitmap doesn't have any bit set, as below:
```
bitmap_foreach_set(b, i) {
    test failed!!! return error code
}
```